core/dbt/cli/main.py

```diff
@@ -52,6 +53,11 @@ def invoke(self, args: List[str]) -> Tuple[Optional[List], bool]:
                 "manifest": self.manifest,
             }
             return cli.invoke(dbt_ctx)
+        except click.exceptions.Exit as e:
+            # 0 exit code, expected for --version early exit
+            if str(e) == "0":
+                return [], True
+            raise e
```

Comment:
can we define our own exception and raise it here? callers of `dbtRunner` (e.g. dbt-server) shouldn't need to import click in order to catch specific exceptions raised during core's execution.